lib/cn/cn_tree.c

```diff
@@ -1589,9 +1623,17 @@ cn_comp_update_split(
             cn_tree_samp(tree, &w->cw_samp_post);
         }
 
+        /* Advance the split generation count to ensure that all spill operations
+         * that overlap this split will abort and retry.
+         */
+        tree->ct_splitgen++;
```

Comment:
The scheduler is in control and should not schedule operations that conflict.

Reply:
Yeah, we can't get there from here.  The scheduler cannot currently accomodate chains of work dependencies, so I'll have to build that in at some point so that we can prevent new root spills while there are pending splits, and prevents splits while there are running rspills.  There cannot be any concurrency unless we can abort/retry a job while holding the tree lock and before transforming the tree...